tests/src/PatternAssertion/LinkPatternAssert.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the link pattern.
+ */
+class LinkPatternAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'label' => [
+        [$this, 'assertElementText'],
+        'a',
+      ],
+      'path' => [
+        [$this, 'assertElementAttribute'],
+        'a',
+        'href',
+      ],
+      'icon' => [
+        [$this, 'assertIcon'],
+      ],
+      'settings' => [
+        [$this, 'assertSettings'],
+      ],
+      'attributes' => [
+        [$this, 'assertAttributes'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+
+    $this->assertElementExists('a', $crawler);
+  }
+
+  /**
+   * Asserts the link icon.
+   *
+   * @param string $expected
+   *   The expected settings.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertIcon(string $expected, Crawler $crawler): void {
+    $svg = $crawler->filter('svg');
+    self::assertCount(1, $svg);
+
+    (new IconPatternAssert())->assertPattern([
+      'name' => $expected,
+      'size' => 's',
+    ], $svg->outerHtml());
+  }
+
+  /**
+   * Asserts the link pattern settings.
+   *
+   * @param array[] $expected
+   *   The expected settings.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertSettings(array $expected, Crawler $crawler): void {
+    if (!empty($expected['disabled'])) {
+      $this->assertElementHasClass('disabled', 'a', $crawler);
+      $this->assertElementAttribute('true', 'a', 'aria-disabled', $crawler);
+    }
+    else {
+      $this->assertElementNotHasClass('disabled', 'a', $crawler);
+      $this->assertElementAttribute(NULL, 'a', 'aria-disabled', $crawler);
+    }
+
+    if (!empty($expected['standalone'])) {
+      $this->assertElementHasClass('standalone', 'a', $crawler);
+    }
+    else {
+      $this->assertElementNotHasClass('standalone', 'a', $crawler);
+    }
+
+    if (isset($expected['icon_position'])) {
+      $link = $crawler->filter('a')->getNode(0);
+
+      // If no icon or no label is set then the position is irrelevant.
+      if (count($link->childNodes) === 2) {
+        $node_name = $expected['icon_position'] === 'before' ? 'svg' : '#text';
+        self::assertSame($node_name, $link->firstChild->nodeName);
```

Comment:
There's a scenario that now I forgot, which doesn't make me like assertSame for scalar assertions. I remember only that it's harder to debug the errors with assert same, I'd rather use assertEquals.

Reply:
ok, I changed also in the TwigExtention test